### ISSUE TITLE
Can't find Python executable when installing Mongoose 3.8.23 on Windows 7 (64-bit)

### ISSUE THREAD

**Issue by hezjing** [LearnBoost/mongoose#2697]
Hi

It looks like mongoose 3.8.23 is installed successfully on Windows 7 (64-bit) when I executed `npm install`
```
>npm install
npm WARN package.json dummy@1.0.0 No description
npm WARN package.json dummy@1.0.0 No repository field.
npm WARN package.json dummy@1.0.0 No README data
/
> kerberos@0.0.7 install c:\dummy\node_modules\mongoose\node_modules\mongodb\node_modules\kerberos
> (node-gyp rebuild 2> builderror.log) || (exit 0)


c:\dummy\node_modules\mongoose\node_modules\mongodb\node_modules\kerberos>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild

> bson@0.2.19 install c:\dummy\node_modules\mongoose\node_modules\mongodb\node_modules\bson
> (node-gyp rebuild 2> builderror.log) || (exit 0)

c:\dummy\node_modules\mongoose\node_modules\mongodb\node_modules\bson>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
mongoose@3.8.23 node_modules\mongoose
├── regexp-clone@0.0.1
├── muri@0.3.1
├── sliced@0.0.5
├── hooks@0.2.1
├── mpath@0.1.1
├── ms@0.1.0
├── mpromise@0.4.3
├── mquery@0.8.0 (debug@0.7.4)
└── mongodb@1.4.28 (readable-stream@1.0.33, kerberos@0.0.7, bson@0.2.19)
```
However, there are `builderror.log` generated in `bson` and `kerberos` directories under `node_modules\mongoose\node_modules\mongodb\node_modules` directory that looks like the following:
```
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at failNoPython (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:103:14)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:64:11
gyp ERR! stack     at FSReqWrap.oncomplete (evalmachine.<anonymous>:99:15)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd c:\c:\dummy\projects\x\node_modules\mongoose\node_modules\mongodb\node_modules\kerberos
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
```
Should I be wary about these errors? Do I need to install Python before installing Mongoose on Windows 7?

**Comment by vkarpov15** [comment]
You don't necessarily need to worry about these error messages (they're actually warnings, since the `kerberos` module is optional and the `bson` library can work without its native extension). Basically, NodeJS has this clunky build tool called `node-gyp` that's the standard way to compile C++ extensions for NodeJS, and it requires python. See the [github repo](https://github.com/TooTallNate/node-gyp) for more info. The kerberos module and the bson module use node-gyp to build.

Long story short, with your current setup, mongoose will work fine unless you want to use kerberos authentication. You will also see a "Failed to load c++ bson extension, using pure JS version" message when you start mongoose. When bson can't compile its c++ extension, it will use a pure JS version, which is somewhat slower but still functional.

**Comment by KodkodGates** [comment]
Thx vkarpov15
